src/ChluIPFS.js

```diff
@@ -205,6 +228,28 @@ class ChluIPFS {
                 // Emit internal REPLICATED event
                 this.events.emit(constants.eventTypes.replicated + '_' + obj.address);
             }
+
+            if (this.type === constants.types.service) {
```

Comment:
@fazo96 I think we should refactor ChluIPFS class to use delegation pattern. Over time these message types and various conditions might grow and we'd end up with a big if/else block with code inside it. Instead with delegation at least we get the code out into separate classes that we can test independently then. What do you think? If you agree, you should create an issue and then we can get to it over the next week.

Reply:
I agree, I opened #16